### PR TITLE
Support for ChromeDriver arguments

### LINKED ISSUE #189: Passing `--user-data-dir` is interpreted as Spectron, not Chrome argument
**Spectron version**: 3.6.2

### Intention 
Working on https://github.com/Microsoft/vscode/issues/25291, I want to be able to pass `--user-data-dir` to my Electron application.

### Problem Description
When instantiating `Application` and passing `--user-data-dir` as an argument to `args`, the argument is not passed as native Chrome argument, but as a Spectron one.

```
new Application({
            path: electronPath,
            args: [cwd, '--user-data-dir=C:\\test-user-dir'],
            env: env
        });
```
This creates a problem that running application instance contains two `--user-data-dir` parameters when spawned, where second one is a fake directory created by WebDriver (because `--user-data-dir` is not supplied and the one specified in constructor is passed with `spectron-arg` prefix to it):

`"C:\Executable.exe" c:\cwd --user-data-dir=C:\test-user-dir --disable-background-networking --disable-client-side-phishing-detection --disable-default-apps --disable-hang-monitor --disable-infobars --disable-popup-blocking --disable-prompt-on-repost --disable-sync --disable-web-resources --enable-logging --ignore-certificate-errors --load-component-extension=C:\Users\t-mikapo\AppData\Local\Temp\scoped_dir15424_30047\internal --log-level=0 --metrics-recording-only --no-first-run --password-store=basic --remote-debugging-port=12730 --safebrowsing-disable-auto-update --test-type=webdriver --use-mock-keychain --user-data-dir=C:\Users\t-mikapo\AppData\Local\Temp\scoped_dir15424_28295 data:,`

See `--user-data-dir=C:\test-user-dir` and ` --user-data-dir=C:\Users\t-mikapo\AppData\Local\Temp\scoped_dir15424_28295` above.

### Verification 
To verify that WebDriver does not create a fake directory I've hardcoded `var args = ['--user-data-dir=C:\test-user-dir']` here:
https://github.com/electron/spectron/blob/master/lib/application.js#L137.

This resulted in Electron app running with the provided '--user-data-dir' and no fake one was created.

### Proposal
Currently Application `args` accept arguments that are passed only to executable. Even API says `See here for details on the Chrome arguments.`, they are not (or not fully with respect to the scenario above) taken into account.

It would be great to have a separate option on `Application` constructor, that allows to pass actual Chrome arguments to it. Please let me know if PR is needed.

### PR DESCRIPTION
**Pull request by michelkaporin**
Added capability of providing ChromeDriver arguments. This resolves #189.

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -144,6 +146,10 @@ Application.prototype.createClient = function () {
       args.push('spectron-env-' + name + '=' + self.env[name])
     }
 
+    for (var index in self.chromeDriverArgs) {
```
Comment (kevinsawicki) [inline lib/application.js]: Might be a little cleaner written in the style of line 141:

```js
self.chromeDriverArgs.forEach(function (arg) {
  args.push(arg)
})
```

File: README.md [hunk 2]
```diff
@@ -74,9 +74,9 @@ Create a new application with the following options:
   script then you should specify `path` as  `electron` via `electron-prebuilt`
   and specify your app's main script path as the first argument in the `args`
   array.
-* `args` - Array of arguments to pass to the executable.
-  See [here](https://sites.google.com/a/chromium.org/chromedriver/capabilities)
-  for details on the Chrome arguments.
+* `args` - Array of arguments to pass to the Electron application.
+* `chromeDriverArgs` - Array of arguments to pass to the ChromeDriver.
```
Comment (kevinsawicki) [inline README.md]: I think this can be just `to ChromeDriver` instead of `to the ChromeDriver.

File: lib/application.js (hunk 1 above)
Comment (michelkaporin) [inline lib/application.js]: Updated it 👍

File: README.md (hunk 2 above)
Comment (michelkaporin) [inline README.md]: Done.

**Comment by michelkaporin** [comment]
@kevinsawicki thanks, I forgot to add typings initially -> added them in 694288e now.

**Comment by kevinsawicki** [comment]
@michelkaporin thanks for this, great work 👍 

I've published this in 3.7.1. The 3.7.x line supports Electron 1.7.x, so if you need this in Electron 1.6.x, let me know and we can merge this into that release line and cut a new 3.6.x spectron release.

**Comment by michelkaporin** [comment]
@kevinsawicki it would be great if you can cut a new 3.6.x spectron release please, as we are on Electron 1.6.6 currently.

**Comment by kevinsawicki** [comment]
> it would be great if you can cut a new 3.6.x spectron release please, as we are on Electron 1.6.6 currently.

Sure thing, published as 3.6.4 via #195

**Comment by michelkaporin** [comment]
Perfect, thank you @kevinsawicki!

**Pull request merged**